### PR TITLE
Add membership api doc

### PR DESCRIPTION
**Pull request by hongu-ku**
In the Messaging API, you can now get membership information.
- add `/membership/v1/subscription/{userId}`
- add `/membership/v1/list`

News: https://developers.line.biz/en/news/2024/02/09/get-membership-plan-informations/

### CODE REVIEW

File: membership.yml [hunk 1]
```diff
@@ -0,0 +1,299 @@
...
+                      joinedTime: 1707214784
+                      nextBillingDate: "2024-02-08"
+                      totalSubscriptionMonths: 1
+
+        "400":
+          description: "An invalid user ID is specified."
+          content:
+            application/json:
+              schema:
+                "$ref": "#/components/schemas/MembershipListResponse"
```
Comment (Yang-33) [inline membership.yml:62]: this should be `ErrorResponse`...?

File: membership.yml [hunk 2]
```diff
@@ -0,0 +1,299 @@
...
+        "404":
+          description: "Unable to get information about the membership to which the user subscribes."
+          content:
+            application/json:
+              schema:
+                "$ref": "#/components/schemas/MembershipListResponse"
```
Comment (Yang-33) [inline membership.yml:68]: (ditto)

File: membership.yml [hunk 3]
```diff
@@ -0,0 +1,299 @@
...
+                    memberCount: 0
+                    memberLimit: null
+                    isInAppPurchase: false
+                    isPublished: true
+        "404":
+          description: "Unable to get information about the memberships."
+          content:
+            application/json:
+              schema:
+                "$ref": "#/components/schemas/ErrorResponse"
```
Comment (Yang-33) [inline membership.yml:112]: (ditto)

File: membership.yml [hunk 4]
```diff
@@ -0,0 +1,299 @@
...
+                "$ref": "#/components/schemas/MembershipListResponse"
+        "404":
+          description: "Unable to get information about the membership to which the user subscribes."
+          content:
+            application/json:
+              schema:
+                "$ref": "#/components/schemas/MembershipListResponse"
+  "/membership/v1/list":
+    get:
+      tags:
```
Comment (Yang-33) [inline membership.yml:71]: can you add `externalDocs.url` for `GET /membership/v1/list`?

**Review by Yang-33 — COMMENTED** [review]
Thank you for adding this~

File: membership.yml (hunk 1 above)
Comment (Yang-33) [reply membership.yml:62]: I haven't tested this. Is the type of the error body in case of a bad request `ErrorResponse`?

File: membership.yml [hunk 5]
```diff
@@ -0,0 +1,299 @@
...
+          description: "Membership plan ID."
+        title:
+          type: string
+          description: "Membership plan name."
+        description:
+          type: string
+          description: "Membership plan description."
+        benefits:
+          type: array
+          minItems: 1
```
Comment (Yang-33) [inline membership.yml:186]: Some validators may error if the number of items in the response is 0. Is there always one or more?
Writing min/max is useful for Messaging API user, but the membershipAPI must always adhere to it.

Comment (hongu-ku) [reply membership.yml:188]: Yes, each membership plan must offer at least one benefit.
However, The maximum number of benefits offered may be extended, so `maxItems` was removed.(https://github.com/line/line-openapi/pull/51/commits/c33eff5027072d746bed1ac26d84e9e9bffff839)

File: membership.yml (hunk 4 above)
Comment (hongu-ku) [reply membership.yml:73]: fixed in https://github.com/line/line-openapi/pull/51/commits/4f2be8ede9d0a875b127521fad040f823c8134f3

File: membership.yml (hunk 2 above)
Comment (hongu-ku) [reply membership.yml]: fixed in https://github.com/line/line-openapi/pull/51/commits/5463a7b8bbaa1bf7a64fbbd579c1243de30b6840

File: membership.yml (hunk 1 above)
Comment (hongu-ku) [reply membership.yml]: fixed in https://github.com/line/line-openapi/pull/51/commits/5463a7b8bbaa1bf7a64fbbd579c1243de30b6840

File: membership.yml (hunk 4 above)
Comment (Yang-33) [reply membership.yml:73]: Thank you~

File: membership.yml (hunk 2 above)
Comment (Yang-33) [reply membership.yml]: Thank you~

**Review by Yang-33 — APPROVED** [review]
LGTM! Thank you for adding this file, @hongu-ku san~
I'll merge this change after I receive comment from you in internal Slack.

**Review by Yang-33 — APPROVED** [review]
👍

**Pull request merged**